crates/subspace-farmer/src/single_disk_farm.rs

```diff
@@ -1105,86 +1105,96 @@ impl SingleDiskFarm {
-                    info!(
+                    if max_pieces_in_sector > pieces_in_sector {
+                        info!(
                         pieces_in_sector,
                         max_pieces_in_sector,
                         "Farm initialized with smaller number of pieces in sector, farm needs to \
                         be re-created for increase"
                     );
-                }
+                    }
 
-                if allocated_space != single_disk_farm_info.allocated_space() {
-                    info!(
-                        old_space = %bytesize::to_string(single_disk_farm_info.allocated_space(), true),
-                        new_space = %bytesize::to_string(allocated_space, true),
-                        "Farm size has changed"
-                    );
+                    if allocated_space != single_disk_farm_info.allocated_space() {
+                        info!(
+                            old_space = %bytesize::to_string(single_disk_farm_info.allocated_space(), true),
+                            new_space = %bytesize::to_string(allocated_space, true),
+                            "Farm size has changed"
+                        );
 
-                    {
-                        let new_allocated_space = allocated_space;
-                        let SingleDiskFarmInfo::V0 {
-                            allocated_space, ..
-                        } = &mut single_disk_farm_info;
-                        *allocated_space = new_allocated_space;
+                        {
+                            let new_allocated_space = allocated_space;
+                            let SingleDiskFarmInfo::V0 {
+                                allocated_space, ..
+                            } = &mut single_disk_farm_info;
+                            *allocated_space = new_allocated_space;
+                        }
+
+                        single_disk_farm_info.store_to(directory)?;
                     }
 
-                    single_disk_farm_info.store_to(directory)?;
+                    (single_disk_farm_info, single_disk_farm_info_lock)
                 }
+                None => {
+                    let single_disk_farm_info = SingleDiskFarmInfo::new(
+                        FarmId::new(),
+                        farmer_app_info.genesis_hash,
+                        public_key,
+                        max_pieces_in_sector,
+                        allocated_space,
+                    );
 
-                single_disk_farm_info
-            }
-            None => {
-                let single_disk_farm_info = SingleDiskFarmInfo::new(
-                    FarmId::new(),
-                    farmer_app_info.genesis_hash,
-                    public_key,
-                    max_pieces_in_sector,
-                    allocated_space,
-                );
-
-                single_disk_farm_info.store_to(directory)?;
+                    single_disk_farm_info.store_to(directory)?;
```

Comment:
It seems that we write first and then secure a lock. Am I missing something?

Reply:
There is no file to lock yet in case we've just created new farm